scalding-core/src/main/scala/com/twitter/scalding/Source.scala

```diff
@@ -39,7 +39,7 @@ import scala.collection.JavaConverters._
 /**
  * thrown when validateTaps fails
  */
-class InvalidSourceException(message: String) extends RuntimeException(message)
+class InvalidSourceException(message: String, cause: Throwable = null) extends RuntimeException(message, cause)
```

Comment:
Yeah, wasn't sure if there's a difference.

It turns out that it's hard to extend Exception in scala because of the difference in how multiple constructors work in scala vs. java. The only hack that makes this possible is that passing null as the throwable just happens to be ok w/ the Exception base class. If that wasn't the case it'd be even worse.